core/controllers/classifier.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Controllers for communicating with the VM for training classifiers."""
+
+from core.controllers import base
+from core.domain import classifier_domain
+from core.domain import classifier_services
+
+import feconf
+import hashlib
+import hmac
+
+def validate_request(handler):
+    """Decorator that checks if the incoming request for storing trained
+    classifier is valid."""
+    def test_is_valid(self, **kwargs):
+        signature = self.request.pop('signature')
+        vm_id = self.request.pop('vm_id')
+        secret = feconf.DEFAULT_VM_SHARED_SECRET
```

Comment:
Nope. This is wrong. Get the secret dynamically from config property of vm_id which I implemented few days ago. So the code should be something like `config_domain.vm_id_config_property.get('vm_id')` (not exact but similar to this one). Check controllers in `admin.py`, they should be using one or more of these config properties and will get you clear idea of how to work with them.